### ISSUE TITLE
flexible-vector-delete! raises an error when removing the last element

### ISSUE THREAD

**Issue by ktakashi** [ktakashi/sagittarius-scheme#262]
**[Original report](https://bitbucket.org/ktakashi/sagittarius-scheme/issues/257) by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

This should not raise an error:

```scheme
(define fv (make-flexible-vector 1 'ok))
(flexible-vector-delete! fv 0)
```

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

Fixing one element deletion of flexible vector (Fixes #257)

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

This should not raise an error:

```scheme
(define fv (make-flexible-vector 1 'ok))
(flexible-vector-delete! fv 0)
```
